pylearn2/models/mlp.py

```diff
@@ -3059,6 +3079,14 @@ def set_input_space(self, space):
 
         rng = self.mlp.rng
 
+        if self.pad != (0, 0):
+            output_shape = \
+                [int(np.ceil((i_sh + 2. * k_pad - k_sh) / float(k_st))) + 1
+                 for i_sh, k_sh, k_st, k_pad in izip(self.input_space.shape,
+                                                     self.kernel_shape,
+                                                     self.kernel_stride,
+                                                     self.pad)]
+
         if self.border_mode == 'valid':
```

Comment:
(I got rid of pad, following Fred's suggestion, so this comment does not apply anymore to this PR. I keep it for future reference.)

`border_mode` and `pad` can be passed together, as the latter augments the data but does not guarantee to do it such as the final size is a multiple of the kernel convolution (i.e., you might still need to either pad more or ignore part of the _padded image_)